engine/src/components/component.ts

```diff
@@ -0,0 +1,41 @@
+import { Data } from "@xgovformbuilder/model";
+
+import { ViewModel } from "./types";
+
+export default class Component {
+  // TODO: types
+  name: any;
+  title: string | undefined; // TODO: remove undefined fix in constructor (remove code smell Object.assign(this, def))
+  values: any;
+  model: any;
+  options: any;
+  itemValues: any;
+  formSchema: any;
+  stateSchema: any;
+  hint: string | undefined;
+  schema: {
+    max?: number | string;
+    error?: any;
+    regex?: string;
+    precision?: number;
+    email?: boolean;
+  };
+  content: any;
+  items: any;
+
+  constructor(def, model) {
+    // TODO explicitly assign to proper initialize props expected
+    Object.assign(this, def);
+    this.model = model;
+    const data = new Data(model.def);
+    const values = data.valuesFor(def);
+    this.values = values ? values.toStaticValues() : undefined;
```

Comment:
I know this wasn't you, so no need to action, we can use values?.toStaticValues() which should return undef if values is